### PR TITLE
Allow Travis to stage release build

### PR DESCRIPTION
**Pull request by kroikie**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,11 @@
 language: java
 script:
 - ant -f ./client-libraries/java/rest-client/build.xml -lib ./client-libraries/java/rest-client/lib
-  deploy
+  stage
```
Comment (silvolu) [inline .travis.yml]: shouldn't we execute this only on tag?

File: .travis.yml [hunk 2]
```diff
@@ -1,15 +1,18 @@
 language: java
 script:
-- ant -f ./client-libraries/java/rest-client/build.xml -lib ./client-libraries/java/rest-client/lib
-  deploy
+- ./build.sh
 before_install:
+- openssl aes-256-cbc -K $encrypted_8864fe5d711d_key -iv $encrypted_8864fe5d711d_iv
```
Comment (silvolu) [inline .travis.yml]: why in before_install?

File: .travis.yml [hunk 3]
```diff
@@ -1,15 +1,18 @@
...
+  -in sign.key.enc -out sign.key -d
+- gpg --import sign.key
 - cp settings.xml ${HOME}/.m2/settings.xml
-after_success:
-- echo "deploy completed."
 branches:
   only:
   - master
+  - /^v[0-9]{1,2}.[0-9]{1,2}.[0-9]{1,3}$/
```
Comment (silvolu) [inline .travis.yml]: I'm still not sure why we would want to do this.

File: build.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+tagCount=$(git tag -l --points-at HEAD | wc -l)
```
Comment (silvolu) [inline build.sh]: Why not using travis env variable TRAVIS_TAG?

File: .travis.yml (hunk 2 above)
Comment (kroikie) [inline .travis.yml]: That was automatically added there by travis, when the key was encrypted.

File: build.sh [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+if [ $tagCount -gt 0 ]; then
+  echo "tagged so deploy"
+  ant -f ./client-libraries/java/rest-client/build.xml -lib ./client-libraries/java/rest-client/lib deploy
+else
+  echo "not tagged so stage"
```
Comment (silvolu) [inline build.sh]: would be cleaner to no op if it's a pull request, e.g.
"${TRAVIS_PULL_REQUEST}" == "false"

File: .travis.yml (hunk 3 above)
Comment (kroikie) [inline .travis.yml]: Ah! this should be removed to have the build run on all pushes.

File: .travis.yml (hunk 2 above)
Comment (silvolu) [inline .travis.yml]: You can move it to the script, to execute it only when we actually need it.

Comment (kroikie) [inline .travis.yml]: Done

File: build.sh (hunk 5 above)
Comment (kroikie) [inline build.sh]: Done

File: build.sh (hunk 4 above)
Comment (kroikie) [inline build.sh]: Done

File: build.sh [hunk 6]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
Comment (silvolu) [inline build.sh:1]: Nit: add some comments to explain the flow so that it's clear for everyone, then looks ready :)

Comment (kroikie) [inline build.sh:1]: Done

**Pull request merged**